app/models/polling/cert_status_updater.rb

```diff
@@ -0,0 +1,41 @@
+class CertStatusUpdater
+  def initialize(hub_config_api)
+    @hub_config_api = hub_config_api
+  end
+
+  def update_hub_usage_status_for_cert(certificate_to_check)
+    outcomes = entity_ids_for(certificate_to_check).map { |entity_id| cert_is_in_use_for_entity_id?(entity_id, certificate_to_check) }
+
+    if outcomes.all?
+      update_cert_status_for(certificate_to_check)
+    end
+  end
+
+private
+
+  def entity_ids_for(certificate)
+    if certificate.component.type == COMPONENT_TYPE::MSA_SHORT
+      [certificate.component.entity_id]
+    else
+      certificate.component.services.map(&:entity_id)
+    end
+  end
+
+  def cert_is_in_use_for_entity_id?(entity_id, certificate_to_check)
+    certs_in_use = get_certs_from_hub(entity_id, certificate_to_check.encryption?)
+    certs_in_use.include?(certificate_to_check.value)
+  end
+
+  def get_certs_from_hub(entity_id, is_for_encryption)
```

Comment:
I don't think we need this method. If we want to rely on an array, we can just modify the existing method to return the encrytion cert in an array (the signing ones are already an array)

Reply:
The purpose of this method is to call the correct method on the hub config API based on the type of certificate; the array wrapping is just an incidental detail.  I don't want to do this ugly `if` thing inline.

Reply:
why do you need the `if` at all? if we change the hub logic to always return an array, even for encryption, then we don't need any concept of encryption vs signing here..

Reply:
Ah sorry, I misunderstood.  You're right, I'll change it.

Reply:
Wait, no, you do still need the if.  There are two different HubConfigApi methods to get the cert from, depending on its usage.  Combining these into one would require you to pass the usage as a parameter, which feels uglier to me than having two separate methods.  Unless you combine them into one, you still need the if.

Reply:
ah yeah, i forgot there were two methods 🤦‍♂  ... yep, we need the if, but we might not need any array wrapping and transformation

Reply:
The array wrapping is needed, but could be moved to HubConfigApi.